velox/expression/FunctionSignature.h

```diff
@@ -328,9 +399,55 @@ class AggregateFunctionSignatureBuilder {
 
   AggregateFunctionSignatureBuilder& knownTypeVariable(
       const std::string& name) {
-    addVariable(
-        variables_,
-        SignatureVariable(name, "", ParameterType::kTypeParameter, true));
+    if (variables_.find(name) == variables_.end()) {
```

Comment:
Looks like this logic is to allow calling both orderableTypeVariable and knownTypeVariable in the same signature. Do we need this? How about we don't allow if and if necessary add knownOrderableTypeVariable later?

Reply:
It seems we do not need it for now. It is better to make them mutually exclusive and keep `SignatureVarible` immutable. I will update this PR according to these principles.